osquery/events/darwin/diskarbitration.h

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <CoreServices/CoreServices.h>
+#include <IOKit/IOKitLib.h>
+#include <DiskArbitration/DiskArbitration.h>
+
+#include <boost/thread/locks.hpp>
+#include <boost/thread/mutex.hpp>
+#include <boost/make_shared.hpp>
+
+#include <osquery/events.h>
+#include <osquery/status.h>
+
+namespace osquery {
+
+#define kIOPropertyProtocolCharacteristicsKey_ "Protocol Characteristics"
+#define kVirtualInterfaceLocation_ "Virtual Interface Location Path"
+#define kDAMediaEjectable_ "DAMediaEjectable"
+#define kDAMediaBSDName_ "DAMediaBSDName"
+#define kDAMediaSize_ "DAMediaSize"
+#define kDAAppearanceTime_ "DAAppearanceTime"
+#define kDADeviceModel_ "DADeviceModel"
+#define kDAVolumeMountable_ "DAVolumeMountable"
+#define kDAVolumeWriteable_ "DAVolumeWriteable"
+#define kDAMediaName_ "DAMediaName"
+#define kDAMediaBlockSize_ "DAMediaBlockSize"
+#define kDADeviceVendor_ "DADeviceVendor"
+#define kDAMediaRemoveable_ "DAMediaRemoveable"
+#define kDAMediaContent_ "DAMediaContent"
+#define kDAVolumeUUID_ "DAVolumeUUID"
+#define kDAVolumeName_ "DAVolumeName"
+#define kDAVolumeKind_ "DAVolumeKind"
+
+const std::string kIOHIDXClassPath_ =
+    "IOService:/IOResources/IOHDIXController/";
```

Comment:
Now that I look at this, it's a bit obtuse for someone to stick as a restriction into a subscription context. It might be better to have a limiter for just DMG events as a bool in the subscription context. If the bool is set then the `starts_with` check is applied.